### PR TITLE
Job template generator

### PR DESCRIPTION
**Pull request by BlaineIrvin**
## Why?

Add generator for creating a basic pdf generation job template and queue configuration.

## What Changed

* [x] Add rails for railtie and generator
* [x] Change generator file structure
* [x] Add job and initializer templates
* [x] Create generator test

### CODE REVIEW

File: lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  include Chromium::Pdf
+
+  queue_as :pdf
+  retry_on Errno::ENOENT, wait: :polynomially_longer, attempts: 3
+  retry_on JobTimeoutError, wait: :polynomially_longer, attempts: 3
+
+  good_job_control_concurrency_with(
+    enqueue_limit: 1,
+    perform_limit: 1,
+    key: -> { "generate_report_pdf_#{arguments.first.id}" }
```
Comment (mikehale) [inline lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb:15]:
```suggestion
    key: -> { arguments.first.id }
```

File: lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class GenerateReportPdfJob < ApplicationJob
```
Comment (mikehale) [inline lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb:3]:
```suggestion
class GeneratePdfJob < ApplicationJob
```

File: lib/generators/chromium/pdf/install/install_generator.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'rails/generators/base'
+require 'rails/generators/active_record'
```
Comment (mikehale) [inline lib/generators/chromium/pdf/install/install_generator.rb:4]:
```suggestion
```

File: lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def perform(model)
+    # Any pre-generation logic
+    return if model.attachment.attached?
+
+    generate_pdf!(model.filename, model.print_url) do |file, filename|
+      # What to do with the generated file
+      model.attachment.attach(io: File.open(file.path), filename: filename, content_type: 'application/pdf')
+      model.save
```
Comment (mikehale) [inline lib/generators/chromium/pdf/install/templates/app/jobs/generate_pdf_job.rb:31]: is save needed?

**Review by OutlawAndy — APPROVED** [review]

**Pull request merged**